src/libraries/Common/src/System/HexConverter.cs

```diff
@@ -275,13 +275,13 @@ public static bool TryDecodeFromUtf16(ReadOnlySpan<char> chars, Span<byte> bytes
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static int FromChar(int c)
         {
-            return c >= CharToHexLookup.Length ? 0xFF : CharToHexLookup[c];
+            return (uint)c >= 'f' + 1 ? 0xFF : CharToHexLookup[c];
```

Comment:
```csharp
(uint)c > 'f' ? 0xFF : CharToHexLookup[c];
```

Reply:
This would make sense if the size of CharToHexLookup could be reduced. Otherwise, what is the benefit of this optimization?

Reply:
The JIT seems to require the GE operand to perform the elimination.

https://csharp.godbolt.org/z/Wb4MqsWYP

Reply:
> This would make sense if the size of CharToHexLookup could be reduced. Otherwise, what is the benefit of this optimization?

We already do this for `FromUpperChar`. I'm assuming it's beneficial avoiding an unnecessary load?

https://github.com/dotnet/runtime/blob/9b76c28567640e4cbe0d20e18b765b8f1a47473f/src/libraries/Common/src/System/HexConverter.cs#L284

Reply:
Filed https://github.com/dotnet/runtime/issues/81160

Reply:
I mean CharToHexLookup size could be reduced to 128.

Reply:
> I mean CharToHexLookup size could be reduced to 128.

It doesn't look so - there are places where it's accessed without cheking bounds using a byte indexer.

Reply:
I have seen a whole series of PRs trying to reduce the size of this library. So the question is whether it's important enough to add one command but reduce the size.